src/lib/sdk/postprocessing-pipeline-elements.ts

```diff
@@ -109,14 +109,12 @@ export abstract class BaseStreamingPostProcessingPipelineElement<TRequestModel,
      * @returns The ResponseModel or ErrorModel returned, based on processing of the DTO.
      */
      processGatewayResponse(responseModel: TResponseModel, dto: TDTO): TResponseModel | TErrorModel {
-        if (dto.status === 'success') {
-            const data = this.transformResponseModel(responseModel, dto)
-            const status = data.status
-            if (status === 'success') {
-                return data as TResponseModel
-            }
-            return data as TErrorModel
+        const data = this.transformResponseModel(responseModel, dto)
+        const status = data.status
+        if (status === 'success') {
+            return data as TResponseModel
```

Comment:
These are the changes that allow the subscriptions to be optional. Basically, the transformResponseModel function is now responsible for determining whether the postprocessing pipeline element output is optional.